libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1755,23 +1938,39 @@ string YulUtilFunctions::updateStorageValueFunction(
 		else
 		{
 			auto const* toReferenceType = dynamic_cast<ReferenceType const*>(&_toType);
-			auto const* fromReferenceType = dynamic_cast<ReferenceType const*>(&_toType);
+			auto const* fromReferenceType = dynamic_cast<ReferenceType const*>(&_fromType);
 			solAssert(fromReferenceType && toReferenceType, "");
 			solAssert(*toReferenceType->copyForLocation(
 				fromReferenceType->location(),
 				fromReferenceType->isPointer()
 			).get() == *fromReferenceType, "");
+			solAssert(fromReferenceType->location() != DataLocation::Storage, "");
```

Comment:
solUnimplementedAssert?

Reply:
In case of `storage` to `storage` it should just do `slot := value`? I think this is done in other place. In any case I can change this to unimplemented assert

Reply:
Please keep it a proper assert if it cannot be triggered - it was just a question.